.github/workflows/build_workflow.yml

```diff
@@ -0,0 +1,340 @@
+name: GP2GP Build Workflow
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  checkstyle:
+    name: Checkstyle
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Java 21 LTS
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: 'temurin'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Create Temporary Artifacts Directory
+        run: mkdir artifacts
+
+      - name: Checkstyle
+        run: |
+          ./gradlew checkStyleMain
+          ./gradlew checkstyleTest
+        working-directory: ./service
+
+      - name: Collect Artifacts
+        run: cp -r ./service/build/reports ./artifacts
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: 'Checkstyle Reports'
+          path: ./artifacts/**
+          compression-level: 9
+
+      - name: Delete Temporary Artifacts
+        run: rm -r ./artifacts
```

Comment:
Could we use "rm -rf" as in automated environments it is a more preferable options.
If for any reason, "./artifacts" don't exist, the command "rm -r" will fail, "rm -rf" is more tolerant and will carry on.